src/inmanta/agent/executor.py

```diff
@@ -341,26 +325,41 @@ async def create_environment(self, blueprint: EnvBlueprint, threadpool: ThreadPo
         """
         env_storage, is_new = self.get_or_create_env_directory(blueprint)
         process_environment = ExecutorVirtualEnvironment(env_storage, threadpool)
-        inmanta_env_status = os.path.join(env_storage, ".inmanta_env_status")
+        inmanta_env_status = pathlib.Path(env_storage) / ".inmanta_env_status"
 
-        if not is_new:
-            current_status = self.read_environment_status(inmanta_env_status)
+        def reset_virtual_environment(current_env_storage: str) -> None:
+            shutil.rmtree(current_env_storage)
+            os.makedirs(env_storage)
 
-            match current_status:
-                case VirtualEnvStatus.created | VirtualEnvStatus.running:
-                    pass
-                case _:
-                    shutil.rmtree(env_storage)
-                    os.makedirs(env_storage)
-                    is_new = True
+        loop = asyncio.get_running_loop()
+
+        if not is_new:
+            if not inmanta_env_status.exists():
+                LOGGER.info(
+                    "Removing venv for content %s, content hash: %s located in",
+                    str(blueprint),
+                    blueprint.blueprint_hash(),
+                    env_storage,
+                )
+                await loop.run_in_executor(threadpool, reset_virtual_environment, env_storage)
+                is_new = True
 
         if is_new:
             LOGGER.info("Creating venv for content %s, content hash: %s", str(blueprint), blueprint.blueprint_hash())
-            await process_environment.create_and_install_environment(blueprint)
+
+            def async_func_wrapper(current_blueprint: EnvBlueprint) -> None:
```

Comment:
I would not do this, I would either move the threadpool deeper down or make this method sync, but this is a bit too complex for comfort